pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -1019,6 +1020,166 @@ func TestHandlePodResourcesResizeWithSwap(t *testing.T) {
+			admitFunc: func(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+				podResources := attrs.Pod.Spec.Containers[0].Resources.Requests
+				if (attrs.Pod.UID == basePod1.UID) && reflect.DeepEqual(podResources, cpu2Mem2G) {
+					return lifecycle.PodAdmitResult{Admit: false, Reason: "Test rejecting high resources"}
+				}
+				return lifecycle.PodAdmitResult{Admit: true}
+			},
+			// pod is still admitted as allocated resources are considered during admission.
+			expectAdmit: true,
+			//  allocated Resources state must not be updated.
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{basePod1.UID: cpu1Mem1G, basePod2.UID: cpu1Mem1G},
+		},
+		{
+			name:                           "allocated resources updated when pod is admitted",
+			initialAllocatedResourcesState: map[types.UID]v1.ResourceList{basePod1.UID: cpu1Mem1G},
+			activePods:                     []*v1.Pod{pod1v1, pod2v1},
+			podToAdd:                       pod2v1,
+			admitFunc:                      nil,
+			expectAdmit:                    true,
+			// pod2's resources added to allocated resources
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{basePod1.UID: cpu1Mem1G, basePod2.UID: cpu1Mem1G},
+		},
+		{
+			name:                           "allocated resources not modified if pod is not admitted",
+			initialAllocatedResourcesState: map[types.UID]v1.ResourceList{basePod1.UID: cpu1Mem1G, basePod2.UID: cpu1Mem1G},
+			activePods:                     []*v1.Pod{pod1v1, pod2v1},
+			podToAdd:                       pod1v2,
+			admitFunc: func(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+				if attrs.Pod.UID == basePod1.UID {
+					return lifecycle.PodAdmitResult{Admit: false, Reason: "Test rejecting pod 1"}
+				}
+				return lifecycle.PodAdmitResult{Admit: true}
+			},
+			expectAdmit: false,
+			// pod1's allocated resources not modified
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{basePod1.UID: cpu1Mem1G, basePod2.UID: cpu1Mem1G},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			allocationManager := makeAllocationManager(t, &containertest.FakeRuntime{}, []*v1.Pod{})
+
+			podForAllocation := func(uid types.UID, resources v1.ResourceList) *v1.Pod {
+				return &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{UID: uid},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{{
+							Name:      containerName,
+							Resources: v1.ResourceRequirements{Requests: resources},
+						}},
+					},
+				}
+			}
+
+			for podUID, resources := range tc.initialAllocatedResourcesState {
+				err := allocationManager.SetAllocatedResources(podForAllocation(podUID, resources))
+				require.NoError(t, err)
+			}
+
+			if tc.admitFunc != nil {
+				handler := &testPodAdmitHandler{admitFunc: tc.admitFunc}
+				allocationManager.AddPodAdmitHandlers(lifecycle.PodAdmitHandlers{handler})
+			}
+
+			ok, _, _ := allocationManager.AddPod(tc.activePods, tc.podToAdd)
```

Comment:
Add assertions that the values of `reason` and `message` are what you expect. We should have cases where they are not empty.

Reply:
Added.